numpy/_core/src/multiarray/multiarraymodule.c

```diff
@@ -4586,6 +4588,8 @@ static struct PyMethodDef array_module_methods[] = {
         "Give a warning on reload and big warning in sub-interpreters."},
     {"from_dlpack", (PyCFunction)from_dlpack,
         METH_FASTCALL | METH_KEYWORDS, NULL},
+    {"unique_hash",  (PyCFunction)array_unique,
```

Comment:
NIT:
```suggestion
    {"_unique_hash",  (PyCFunction)array__unique_hash,
```
(or not the double `__`.  I am fine either way, but I think it is good to call it `unique_hash` also internally for clarity.)